### DISCUSSION TITLE
Overridable Acornfiles

### DISCUSSION THREAD

**Discussion by torbenj** [acorn-io/runtime#1380] [Ideas]
I would like to propose an idea as a follow up of my idea in discussion #1379 .

I like the idea to work with the same technology locally that gets used for deployment in production (or on other stages) but oftentimes I have different requirements during development than during production.

Let's imagine I'm working on a project consisting of a backend service and a database. The backend service uses the database and in production the backend service exposes its API on a certain port to the outside. The database itself is only reachable within the cluster i.e. only reachable by the backend service but not from the outside.
During development I want to access the database directly so I have to expose it to localhost.
To achieve this I usually use some docker-compose file that defines the two services and expose the database port to localhost.

Docker-compose has the concept of override files so that the user can define additional docker-compose files that override certain aspects the "main" file or add new things to it. There might be override files for local development that expose certain services that are not exposed in the prod stage, there might be different port assignments, environment variables, etc.
The development team can decide to add these override files to the repository or ignore them making it possible to have local override files with sensitive information or developer-specific overrides that never get into the git repository.

It would be great if acorn would support a similiar mechanism to avoid having to juggle around with if conditions and having to constantly edit the main Acornfile to achieve certain configurations in different scenarios.

**Comment by torbenj** [comment]
The "profiles" feature of acorn was mentioned in slack to resolve the "i would like to have different settings for different stages" issue mentioned in my idea. Although this might be a suitable fix in some way it doesn't resolve the issue that those changes cannot be ignored from being checked in to the repository as I would have to edit the main Acornfile and when I have to pass sensitive information to my services via environment variables I wouldn't want to check them into the repository too :)

**Comment by cjellick** [comment]
Hi @TorbenJ. Another good suggestion. @ibuildthecloud is working on a bit of refactor that introduces a few key concepts: nested acorns and 1st class "services."

I think (though I'm not sure) that your use case might align with what he's doing there. I'll bring this up with him today for consideration and see if he thinks what he is doing will enable your usecase.
